### PR TITLE
Fixed #28044 -- Unified the logic for createsupeuser's interactive and --noinput modes.

### PR DESCRIPTION
**Pull request by nerogit**
https://code.djangoproject.com/ticket/28044

### CODE REVIEW

**Comment by timgraham** [comment]
@nerogit, I rebased this branch and added some tests from https://github.com/django/django/pull/9554. Also, the branch needs some adaptions to fix the test added in 6aec130a4c19ed17a61835af15b4fb2186dfacad. Do you want to try fixing the TODO comments in the patch?

**Comment by nerogit** [comment]
I see, you means 
https://github.com/django/django/pull/8908/commits/b5c00e42798cbea75489ddf3dde751e5c6ead91d#diff-f9f4f9f5c558be1dd8e84be93a3cf3c2R553

and

https://github.com/django/django/pull/8908/commits/b5c00e42798cbea75489ddf3dde751e5c6ead91d#diff-f9f4f9f5c558be1dd8e84be93a3cf3c2R691

is it?

**Comment by timgraham** [comment]
Yes

File: tests/auth_tests/test_management.py [hunk 1]
```diff
@@ -550,7 +550,6 @@ def test(self):
 
         test(self)
 
-    # TODO: fix this test
     def _test_blank_username(self):
```
Comment (timgraham) [inline tests/auth_tests/test_management.py]: You need to remove the underscore prefix from the test name to allow the test to run.

**Comment by nerogit** [comment]
@timgraham Done :)

**Comment by timgraham** [comment]
It seems there's a test isolation issue. If I run `./tests/runtests.py auth_tests.test_management --parallel=1` I get a failure and some errors. If I add `--reverse`, then the tests pass. Can you reproduce that?

**Comment by timgraham** [comment]
Actually, that issue isn't related to this patch. I've reported it in the relevant location: https://code.djangoproject.com/ticket/28757#comment:7.

**Comment by timgraham** [comment]
This is looking good, but it decreases test coverage a bit (and some lines are already untesteed). Could you try to add tests for lines 78-81, 161-2, 208-209 (in a separate pull request)? That will bring test coverage to 100%.

**Comment by nerogit** [comment]
@timgraham Okay i'm doing on https://github.com/nerogit/django/tree/Add-test-createsuperuser, but I have a question. When I finish adding test on that lines, then I PR to `django/django:master`?

**Comment by timgraham** [comment]
You can branch from master. We'll merge those tests first, then merge this PR.

**Pull request merged**